firefly-api/src/models.rs

```diff
@@ -19,3 +22,35 @@ pub mod casper {
         tonic::include_proto!("casper.v1");
     }
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TransferResult {
+    pub cost: u64,
+    pub errored: bool,
+    pub system_deploy_error: Option<String>,
+}
```

Comment:
This representation is more correct and safe:
```suggestion
pub enum TransferResult {
    Transfered {
        pub cost: u64,
    },
    Failed {
        pub system_deploy_error: String,
    }
}
```
Its either a cost or a error and its impossible to misuse it

Reply:
good idea, todo in next tasks